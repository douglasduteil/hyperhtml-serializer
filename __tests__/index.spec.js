//

test('should remove extra hyper comments from the html', () => {
  const actual = require('fs')
    .readFileSync(__dirname + '/actual.html')
    .toString();
  const section = document.createElement('section');
  section.innerHTML = actual;
  expect(section).toMatchSnapshot();
});

test('should only act on HTML strings', () => {
  expect('Some text without any tags.').toMatchSnapshot();
  expect({}).toMatchSnapshot();
  expect([]).toMatchSnapshot();
  expect(0).toMatchSnapshot();
  expect(true).toMatchSnapshot();
  expect(() => {}).toMatchSnapshot();
});
