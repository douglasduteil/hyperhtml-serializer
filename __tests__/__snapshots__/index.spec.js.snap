// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should only act on HTML strings 1`] = `"Some text without any tags."`;

exports[`should only act on HTML strings 2`] = `{}`;

exports[`should only act on HTML strings 3`] = `[]`;

exports[`should only act on HTML strings 4`] = `0`;

exports[`should only act on HTML strings 5`] = `true`;

exports[`should only act on HTML strings 6`] = `[Function]`;

exports[`should remove extra hyper comments from the html 1`] = `
<section>
  <foo-hyper-app>
    
    
    <!-- This is my comment -->
    
    
    <h1>
      Hello hyperHTML !
      
    </h1>
    
    
    <h1>
      Hello world !
      
    </h1>
    
  
  </foo-hyper-app>
</section>
`;
